### PR TITLE
Added directory listing feature

### PR DESCRIPTION
**Pull request by regilero**

### CODE REVIEW

File: FileManager.php [hunk 1]
```diff
@@ -368,6 +370,32 @@ private function unsafeDeduplicate(string $filename): string
...
     }
 
+    /**
+     * Use finder to extract files list from a directory
+     * @param $uri is the directory uri
+     * @param $pattern is an optionnal patteren restriction (like '*.csv')
+     * @param $createDirectoryIfNotExists is False by default, if true the $uri
+     *        directory would be created if not exists
+     */
+    public function ls(string $uri, $pattern='*', $createDirectoryIfNotExists=False): Finder
```
Comment (pounard) [inline FileManager.php]: `$createDirectoryIfNotExists = false` this is not Python :)

Comment (pounard) [reply FileManager.php]: `$pattern = '*'`

File: FileManager.php [hunk 2]
```diff
@@ -368,6 +370,32 @@ private function unsafeDeduplicate(string $filename): string
...
+
+        if (! $this->exists($uri)) {
+            if ($createDirectoryIfNotExists) {
+                $this->mkdir($uri);
+            } else {
+                throw new FileNotFoundException();
+            }
+        }
+        $finder = new Finder();
+        return $finder
```
Comment (pounard) [inline FileManager.php:392]: Please add anew line before the `return` statement.

File: FileManager.php [hunk 3]
```diff
@@ -5,8 +5,10 @@
 namespace MakinaCorpus\FilechunkBundle;
 
 use MakinaCorpus\FilechunkBundle\StreamWrapper\LocalStreamWrapper;
+use Symfony\Component\Finder\Finder;
```
Comment (pounard) [inline FileManager.php:8]: I'm not against this dependency, but please add the composer dependency as well. Try to keep it loose `^3.2 || ^4`.

File: FileManager.php [hunk 4]
```diff
@@ -368,6 +370,32 @@ private function unsafeDeduplicate(string $filename): string
         return $candidate;
     }
 
+    /**
+     * Use finder to extract files list from a directory
+     * @param $uri is the directory uri
```
Comment (pounard) [inline FileManager.php]: Blank line before `@param`, parameters must be typed, `@param string $uri`.

**Review by pounard — CHANGES_REQUESTED** [review]

File: FileManager.php (hunk 1 above)
Comment (regilero) [reply FileManager.php]: done

File: FileManager.php (hunk 4 above)
Comment (regilero) [reply FileManager.php]: done

File: FileManager.php (hunk 3 above)
Comment (regilero) [reply FileManager.php:8]: done

File: FileManager.php [hunk 5]
```diff
@@ -368,6 +370,34 @@ private function unsafeDeduplicate(string $filename): string
...
+     * Use finder to extract files list from a directory
+     * 
+     * @param string $uri is the directory uri
+     * @param string $pattern is an optionnal patteren restriction (like '*.csv')
+     * @param bool $createDirectoryIfNotExists is False by default, if true the $uri
+     *        directory would be created if not exists.
+     */
+    public function ls(string $uri, $pattern = '*', $createDirectoryIfNotExists = false): Finder
+    {
+
```
Comment (pounard) [inline FileManager.php]: Drop this empty line.

**Review by pounard — CHANGES_REQUESTED** [review]

File: FileManager.php (hunk 5 above)
Comment (regilero) [reply FileManager.php]: done

**Comment by pounard** [comment]
Impeccable, merci!

**Pull request merged**